drivers/adc/max9611pmb/max9611.c

```diff
@@ -0,0 +1,458 @@
+ * @brief Get LR bit value
+ * @param dev 		  - The device structure.
+ * @param lr	      - Pointer to store LR bit value
+ * @return            - 0 if successful, error otherwise.
+*******************************************************************************/
+int max9611_get_lr(struct max9611_dev *dev, bool *lr)
+{
+	int ret;
+	uint8_t temp;
+
+	ret = max9611_read(dev, MAX9611_CTR1, 1, &temp);
+	if (ret)
+		return ret;
+
+	*lr = (bool)((temp & MAX9611_LR_MASK) >> no_os_find_first_set_bit(
+			     MAX9611_LR_MASK));
+
+	return 0;
+}
+
+/***************************************************************************//**
+ * @brief Configures MAX9611 delay time
+ * @param dev 		  - The device structure.
+ * @param dtime		  - Delay option
+ * @return            - 0 if successful, error otherwise.
+*******************************************************************************/
+int max9611_set_delay(struct max9611_dev *dev, enum max9611_delay_time dtime)
+{
+	dtime <<= no_os_find_first_set_bit(MAX9611_DTIME_MASK);
+
+	return max9611_reg_update(dev, MAX9611_CTR2, dtime,
+				  MAX9611_DTIME_MASK);
+}
+
+/***************************************************************************//**
+ * @brief Get MAX9611 configured delay time
+ * @param dev 		  - The device structure.
+ * @param dtime	      - Pointer to store delay time read
+ * @return            - 0 if successful, error otherwise.
+*******************************************************************************/
+int max9611_get_delay(struct max9611_dev *dev, enum max9611_delay_time *dtime)
+{
+	int ret;
+	uint8_t temp;
+
+	ret = max9611_read(dev, MAX9611_CTR2, 1, &temp);
+	if (ret)
+		return ret;
+
+	*dtime = (temp & MAX9611_DTIME_MASK) >> no_os_find_first_set_bit(
+			 MAX9611_DTIME_MASK);
+
+	return 0;
+}
+
+/***************************************************************************//**
+ * @brief Configures MAX9611 retry time
+ * @param dev 		  - The device structure.
+ * @param rtime		  - Retry time option
+ * @return            - 0 if successful, error otherwise.
+*******************************************************************************/
+int max9611_set_retry(struct max9611_dev *dev, enum max9611_retry_time rtime)
+{
+	rtime <<= no_os_find_first_set_bit(MAX9611_RTIME_MASK);
+
+	return max9611_reg_update(dev, MAX9611_CTR2, rtime,
+				  MAX9611_RTIME_MASK);
+}
+
+/***************************************************************************//**
+ * @brief Get MAX9611 configured retry time
+ * @param dev 		  - The device structure.
+ * @param rtime	      - Pointer to store retry time read
+ * @return            - 0 if successful, error otherwise.
+*******************************************************************************/
+int max9611_get_retry(struct max9611_dev *dev, enum max9611_retry_time *rtime)
+{
+	int ret;
+	uint8_t temp;
+
+	ret = max9611_read(dev, MAX9611_CTR2, 1, &temp);
+	if (ret)
+		return ret;
+
+	*rtime = (temp & MAX9611_RTIME_MASK) >> no_os_find_first_set_bit(
```

Comment:
ditto

Reply:
Resolved in latest commit using no_os_field_get